src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -691,7 +691,8 @@ public function dumpFile(string $filename, $content)
                 throw new IOException(sprintf('Failed to write file "%s": ', $filename).self::$lastError, 0, null, $filename);
             }
 
-            self::box('chmod', $tmpFile, file_exists($filename) ? fileperms($filename) : 0666 & ~umask());
+            $perms = @fileperms($filename);
+            self::box('chmod', $tmpFile, file_exists($filename) && false !== $perms ? $perms : 0666 & ~umask());
```

Comment:
```suggestion
            self::box('chmod', $tmpFile, file_exists($filename) && false !== $perms = fileperms($filename) ? $perms : 0666 & ~umask());
```

Reply:
This isn't permitted by PHP, `$perms` is undefined in the first branch of the ternary expression with this syntax.